Encryption guide copy fixes (#41819)
rails/rails: guides/source/active_record_encryption.md

You might need to ignore case when querying deterministically encrypted data. There are two options that can help you here.

> You can use the option `:downcase` when declaring the encrypted attribute. This will downcase that content before encrypting it.

Reviewer: ```suggestion
You can use the `:downcase` option when declaring the encrypted attribute to downcase the content before encryption occurs.
```

I think we can make it even more concise! Also, we can avoid using "it," which is sometimes fairly confusing.

Author: I like it! Thank you :)

> You can use the option `:downcase`  when declaring the encrypted attribute. This will downcase that content before encrypting it.

Author: ```suggestion
You can use the option `:downcase` when declaring the encrypted attribute. This will downcase that content before encrypting it.
```

```ruby
class Person
  encrypts :email_address, deterministic: true, downcase: true
end
```

> When using `:downcase` the original case is lost. There might be situations where you need to preserve the original case when reading the value, but you need to ignore the case when querying. For those situations you can use the option `:ignore_case` which requires you to add a new column named `original_<column_name>` to store the content with the case unchanged:

Reviewer: ```suggestion
When using `:downcase` the original case is lost. In some situations, you might need to preserve the original case, but ignore that case when querying. For those situations, you can use the `:ignore_case` option which requires a new column named `original_<column_name>` to store the content with the case unchanged:
```

I think this can also be shortened a bit.

> When using `:downcase` the original case is lost. There might be situations where you need to preserve the original case when reading the value, but you need to ignore the case when querying. For those situations you can use the option `:ignore_case` which requires you to add a new column named `original_<column_name>` to store the content with the case unchanged:

Author: ```suggestion
When using `:downcase`, the original case is lost. In some situations, you might want to ignore the case only when querying, while also storing the original case. For those situations, you can use the option `:ignore_case`. This requires you to add a new column named `original_<column_name>` to store the content with the case unchanged:
```

Author: Based on @jacobherrington's suggestion (thank you!), written as another comment for my suggestion-picking convenience.

I intentionally only made minimal changes because otherwise I could spend forever polishing texts, but fair enough :)
